### ISSUE TITLE
msn-pecan fails to build

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dequis/msn-pecan#15]
```
When I try to build the msn-pecan, it gives an error related to ld. I tried
to build it from source and using the pkgbuild that's in AUR
(http://aur.archlinux.org). I would attach the complete log from the build.

Distro: Arch Linux
Kernel: 2.6.24-ARCH x86_64
gcc version: 4.2.3
ld version: 2.18
Pidgin version: both 2.3.1 and 2.4.0
msn-pecan: both 0.0.8 and 0.0.9


```

Original issue reported on code.google.com by `vale...@gmail.com` on 2 Mar 2008 at 4:35

Attachments:
- [msn-pecan-error](https://storage.googleapis.com/google-code-attachments/msn-pecan/issue-15/comment-0/msn-pecan-error)

**Comment by GoogleCodeExporter** [comment]
```
Same here. I'm using the same version of everything than vales92. The error is 
the
same. Tried to build the AUR's tarball using yaourt to automatize the process, 
but
final product is the same.
```

Original comment by `ruica...@gmail.com` on 2 Mar 2008 at 4:55

**Comment by GoogleCodeExporter** [comment]
```
Do you have a dynamic version of libpurple?

http://www.gentoo.org/proj/en/base/amd64/howtos/index.xml?part=1&chap=3
```

Original comment by `felipe.contreras` on 2 Mar 2008 at 5:02

**Comment by GoogleCodeExporter** [comment]
```
I really don't know. How can i know that? I've read that page, and i've tried 
to done
what's described there, but no results 'till now. 

Sorry for the n00bish :(
```

Original comment by `vale...@gmail.com` on 2 Mar 2008 at 5:33

**Comment by GoogleCodeExporter** [comment]
```
Well, I understand that but I think you should have first asked the arch guys.

You can try the following:
ls -l /usr/lib/libpurple*.so

If you don't get anything, then you don't have a shared library.
```

Original comment by `felipe.contreras` on 2 Mar 2008 at 5:53

**Comment by GoogleCodeExporter** [comment]
```
esquiso@Earth:~$ ls -l /usr/lib/libpurple*.so
lrwxrwxrwx 1 root 18 2008-03-02 04:30 "/usr/lib/libpurple.so" -> 
"libpurple.so.0.4.0"

So, i have the shared library. Should i go talk with the arch devs now?
```

Original comment by `vale...@gmail.com` on 2 Mar 2008 at 6:02

**Comment by GoogleCodeExporter** [comment]
```
I'm on Arch x86, msn-pecan compiles ok here and I have this:

$  ls -l /usr/lib/libpurple*.so
lrwxrwxrwx 1 root root 25 2008-03-02 00:28 /usr/lib/libpurple-client.so ->
libpurple-client.so.0.4.0
lrwxrwxrwx 1 root root 18 2008-03-02 00:28 /usr/lib/libpurple.so -> 
libpurple.so.0.4.0

Looks like you're missing the libpurple-client.
```

Original comment by `vudu.cu...@gmail.com` on 2 Mar 2008 at 6:07

**Comment by GoogleCodeExporter** [comment]
```
I don't think the purple-client is important, at least not for msn-pecan.

Perhaps you can try to build with:
make CFLAGS="-fPIC -DHAVE_LIBPURPLE"

If that works then I'll fix msn-pecan appropriately.
```

Original comment by `felipe.contreras` on 2 Mar 2008 at 6:26

**Comment by GoogleCodeExporter** [comment]
```
i've downloaded the tarball (until now, i've tried only with the pkgbuild) and 
edit
the Makefile to this

CFLAGS=-fPIC -Wall -ggdb -I. -DHAVE_LIBPURPLE $(EXTRA_WARNINGS)

it worked. tyvm for your time :)
```

Original comment by `vale...@gmail.com` on 2 Mar 2008 at 6:33

**Comment by GoogleCodeExporter** [comment]
```
All right, it's fixed in git and should be in 0.0.10 in a couple of minutes.
```

Original comment by `felipe.contreras` on 2 Mar 2008 at 9:18
- Changed state: **Verified**
